### PR TITLE
OSDOCS-2186: AliCloud Disk CSI Operator (GA)

### PR DESCRIPTION
**Pull request by lpettyjo**
4.10+

[OSDOCS-2186:](https://issues.redhat.com/browse/OSDOCS-2186) AliCloud Disk CSI Operator added as GA feature.

**Preview**:
https://deploy-preview-34301--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi-alicloud-disk.html
https://deploy-preview-34301--osdocs.netlify.app/openshift-enterprise/latest/storage/container_storage_interface/persistent-storage-csi.html#csi-drivers-supported_persistent-storage-csi

**PTAL**: @jsafrane @ropatil010

### CODE REVIEW

File: storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+toc::[]
+
+== Overview
+
+{product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for Alibaba AliCloud Disk Storage.
+
+Familiarity with xref:../../storage/understanding-persistent-storage.adoc#understanding-persistent-storage[persistent storage] and xref:../../storage/container_storage_interface/persistent-storage-csi.adoc#persistent-storage-csi[configuring CSI volumes] is recommended when working with a CSI Operator and driver.
+
+To create CSI-provisioned PVs that mount to AliCloud Disk storage assets, {product-title} installs the AliCloud Disk CSI Driver Operator and the AliCloud Disk CSI driver by default in the `openshift-cluster-csi-drivers` namespace.
```
Comment (mjpytlak) [inline storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc:14]:
```suggestion
To create CSI-provisioned PVs that mount to AliCloud Disk storage assets, {product-title} installs the AliCloud Disk CSI Driver Operator and the AliCloud Disk CSI driver, by default, in the `openshift-cluster-csi-drivers` namespace.
```

File: modules/persistent-storage-csi-tp-enable.adoc [hunk 2]
```diff
@@ -2,27 +2,32 @@
 //
 // persistent-storage-csi-vsphere.adoc
 // persistent-storage-csi-azure.adoc
+// persistent-storage-csi-alicloud-disk.adoc
```
Comment (mjpytlak) [inline modules/persistent-storage-csi-tp-enable.adoc:5]: While listed here, I do not see this module included in that assembly.

File: storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+* The _AliCloud Disk CSI Driver Operator_ provides a storage class (`alicloud-disk`) that you can use to create persistent volume claims (PVCs). The AliCloud Disk CSI Driver Operator supports dynamic volume provisioning by allowing storage volumes to be created on-demand, eliminating the need for cluster administrators to pre-provision storage.
```
Comment (mjpytlak) [inline storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc:16]: I noticed that your removed the initial capitalization from Driver Operator in modules/persistent-storage-csi-enable. Should this and any other instances also be fixed?

**Review by mjpytlak — CHANGES_REQUESTED** [review]

File: modules/persistent-storage-csi-tp-enable.adoc [hunk 4]
```diff
@@ -2,27 +2,32 @@
...
 endif::[]
+ifeval::["{context}" == "persistent-storage-csi-alicloud-disk"]
+:alicloud:
+endif::[]
 
 [id="csi-tp-enable_{context}"]
-= Enabling the {FeatureName} CSI Driver Operator
-To enable the {FeatureName} Container Storage Interface (CSI) Driver Operator, you must enable feature gates with the `TechPreviewNoUpgrade` feature set.
+= Enabling the {FeatureName} CSI driver operator
+To enable the {FeatureName} Container Storage Interface (CSI) driver operator, you must enable feature gates with the `TechPreviewNoUpgrade` feature set.
```
Comment (sagidlow) [inline modules/persistent-storage-csi-tp-enable.adoc:20]:
```suggestion
To enable the {FeatureName} Container Storage Interface (CSI) driver Operator, you must enable feature gates with the `TechPreviewNoUpgrade` feature set.
```

Operator should be capitalized. 
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-capitalization

File: storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
+[id="persistent-storage-csi-alicloud-disk"]
+= AliCloud Disk CSI Driver Operator
```
Comment (sagidlow) [inline storage/container_storage_interface/persistent-storage-csi-alicloud-disk.adoc:2]:
```suggestion
= AliCloud Disk CSI driver Operator
```
Maintaining consistency with above module. If you decide to keep drive as capitalized, please update with module above with the updated capitalization.

**Review by sagidlow — COMMENTED** [review]
Just a few nit comments.

File: modules/persistent-storage-csi-tp-enable.adoc (hunk 4 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-tp-enable.adoc:20]: This file is actually not being used. It was going to be originally, but it is now out of scope.

File: modules/persistent-storage-csi-tp-enable.adoc [hunk 6]
```diff
@@ -75,6 +79,19 @@ azure-disk-csi-driver-node-mqdhh                        3/3     Running 0
...
+NAME                                                    READY   STATUS  RESTARTS    AGE
+vmware-alicloud-csi-driver-controller-5646dbbf54-cnsx7   9/9     Running 0          4h29m
+vmware-alicloud-csi-driver-node-ch22q                    3/3     Running 0          4h37m
+vmware-alicloud-csi-driver-node-gfjrb                    3/3     Running 0          4h37m
+vmware-alicloud-csi-driver-node-ktlmp                    3/3     Running 0          4h37m
+vmware-alicloud-csi-driver-node-lgksl                    3/3     Running 0          4h37m
+vmware-alicloud-csi-driver-node-vb4gv                    3/3     Running 0          4h37m
+vmware-alicloud-csi-driver-operator-7c7fc474c-p544t      1/1     Running 0          4h29m
+----
+endif::alicloud[]
```
Comment (jsafrane) [inline modules/persistent-storage-csi-tp-enable.adoc:94]: TechPreviewNoUpgrade is not used in Alibaba, do we still need this chunk? Or any change in modules/persistent-storage-csi-tp-enable.adoc in general?

File: modules/persistent-storage-csi-tp-enable.adoc [hunk 7]
```diff
@@ -102,3 +118,12 @@ managed-csi                 disk.csi.azure.com              Delete          Wait
...
 endif::azure[]
+ifdef::alicloud[]
+[source,terminal]
+----
+NAME            PROVISIONER                     RECLAIMPOLICY   VOLUMEBINDINGMODE       ALLOWVOLUMEEXPANSION    AGE
+thin (default)  kubernetes.io/vsphere-volume    Delete          Immediate               false                   5h43m
+thin-csi        csi.alicloud.vmware.com         Delete          WaitForFirstConsumer    true                    4h38m <1>
+----
+<1> AliCloud storage class
+endif::alicloud[]
```
Comment (jsafrane) [inline modules/persistent-storage-csi-tp-enable.adoc:129]: Same here, TechPreviewNoUpgrade is not needed

File: modules/persistent-storage-csi-tp-enable.adoc (hunk 6 above)
Comment (lpettyjo) [reply modules/persistent-storage-csi-tp-enable.adoc:94]: This module is not actually included in the topic. The changes here got "caught" in this PR by accident.

**Comment by jsafrane** [comment]
lgtm

**Pull request merged**